charts/k8s-service/templates/canarydeployment.yaml

```diff
@@ -0,0 +1,325 @@
+{{- /*
```

Comment:
Not for this iteration, but would be nice if in the future we can DRY this up so that the common configurations between this deployment and the main one come from a single place.

Reply:
Indeed - I was thinking the same

Reply:
Need to figure out the cleanest way to do that

Reply:
Probably using `define` and `include` for each of the subblocks: https://v2.helm.sh/docs/chart_template_guide/#the-include-function

Reply:
Opened an issue capturing this improvement for the future - hopefully that is helpful / a good place to put it: https://github.com/gruntwork-io/helm-kubernetes-services/issues/55